### PR TITLE
[FIX] Correction of the observed mass if isotope error is given

### PR DESCRIPTION
**Pull request by sophiamersmann**
If an isotope error is given (e.g. when using the search engine MSGF+), the calulated mass is corrected by subtracting `(isotopeError * protonMass) / charge`.

### CODE REVIEW

**Comment by jpfeuffer** [comment]
Related to #3466

**Comment by timosachsenberg** [comment]
thanks! did you check if different search engines: e.g., Comet, MSGFPlus and our SimpleSearchEngine write use the same MetaValue key? and also numbering scheme (e.g., one could use 1, the other -1)

**Comment by timosachsenberg** [comment]
Hmm did not find the meta value for comet. Can check tomorrow, maybe we don’t parse it?

**Comment by timosachsenberg** [comment]
we also use "isotope_error" error and use 0 for the monoisotopic peak and 1 if the experimental precursor annotation points to the first isotopic peak

**Comment by timosachsenberg** [comment]
I am not 100% sure that the caluclation is correct.
calc_mass as in this code does not need any isotopic correction as it is calculated from the theoretical composition. The actual istopic misassignment error is on the experimental precursor mass.

**Comment by sophiamersmann** [comment]
You are absolutely right about calc_mass / exp_mass. I updated the code - exp_mass is now corrected rather than calc_mass.

**Comment by sophiamersmann** [comment]
I will need to check if different search engines use the same MetaValue key and numbering scheme.

File: src/topp/PercolatorAdapter.cpp [hunk 1]
```diff
@@ -416,7 +417,12 @@ class PercolatorAdapter :
         
         double calc_mass = hit.getSequence().getMonoWeight(Residue::Full, charge)/charge;
         hit.setMetaValue("CalcMass", calc_mass);
-        
+
+        if (hit.metaValueExists("IsotopeError"))
+        {
+          float isoErr = stof(hit.getMetaValue("IsotopeError").toString());
```
Comment (timosachsenberg) [inline src/topp/PercolatorAdapter.cpp]: stof uses the current locale and might thus not work for files produced on a system with a different locale.
use the toString().toFloat() instead (if toFloat() doesn't work)

**Pull request merged**